From 1a840def6a952484ad61949154c2f108984f5f27 Mon Sep 17 00:00:00 2001
From: Rathees Kumar R Chinannan <quic_rrchinan@quicinc.com>
Date: Tue, 21 Nov 2023 22:46:53 +0530
Subject: [PATCH] hostapd: Avoid NULL pointer dereference.

Added NULL check before accessing hapd ptr on hostapd_wpa_event function.

This NULL check is added to avoid unusual hostapd crashes observed
during module reload test cases.

Signed-off-by: Rathees Kumar R Chinannan <quic_rrchinan@quicinc.com>
Change-Id: Ib1adef09434938c0b4483eb91ce23d076bcce0a5
---
 src/ap/drv_callbacks.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/ap/drv_callbacks.c b/src/ap/drv_callbacks.c
index 06dbe84..a33a626 100644
--- a/src/ap/drv_callbacks.c
+++ b/src/ap/drv_callbacks.c
@@ -2169,10 +2169,13 @@ void hostapd_wpa_event(void *ctx, enum wpa_event_type event,
 	struct hostapd_data *hapd = ctx;
 	int b, err;
 	struct hostapd_data *link_bss;
-
 #ifndef CONFIG_NO_STDOUT_DEBUG
 	int level = MSG_DEBUG;
+#endif /* CONFIG_NO_STDOUT_DEBUG */
 
+	if (hapd == NULL)
+		return;
+#ifndef CONFIG_NO_STDOUT_DEBUG
 	if (event == EVENT_RX_MGMT && data->rx_mgmt.frame &&
 	    data->rx_mgmt.frame_len >= 24) {
 		const struct ieee80211_hdr *hdr;
-- 
2.42.0

